cylc/flow/scripts/lint.py

```diff
@@ -420,7 +579,7 @@ def get_reference_text(checks):
 def get_option_parser() -> COP:
     parser = COP(
         COP_DOC,
-        argdoc=[COP.optional(('DIR ...', 'Directories to lint'))],
+        argdoc=[COP.optional(WORKFLOW_ID_OR_PATH_ARG_DOC)],
```

Comment:
The `main` function is not set up to handle workflow IDs, however. It is treating the ID as a relative path. Also I get traceback following this
```
$ cylc lint foo
2022-09-13T14:49:30+01:00 WARNING - Path foo does not exist.
Traceback (most recent call last):
  File "~/.conda/envs/cylc8/bin/cylc", line 33, in <module>
    sys.exit(load_entry_point('cylc-flow', 'console_scripts', 'cylc')())
  File "~/github/cylc-flow/cylc/flow/scripts/cylc.py", line 657, in main
    execute_cmd(command, *cmd_args)
  File "~/github/cylc-flow/cylc/flow/scripts/cylc.py", line 284, in execute_cmd
    entry_point.resolve()(*args)
  File "~/github/cylc-flow/cylc/flow/terminal.py", line 229, in wrapper
    wrapped_function(*wrapped_args, **wrapped_kwargs)
  File "~/github/cylc-flow/cylc/flow/scripts/lint.py", line 718, in main
    f'Checked {target} against {check_names} rules and '
UnboundLocalError: local variable 'check_names' referenced before assignment
```